tutorials/diy-module.md

```diff
@@ -0,0 +1,390 @@
+
+<h1 style="text-align:center;"><img src="https://s3.amazonaws.com/technicalmachine-assets/tutorials/diy-module-guide/servo_jack_identified.jpg" /></h1>
+
+<p style="text-align:center;"><em>DC Barrel Jack on the Servo Module</em></p>
+
+From the [servo module schematic][servo_schematic], we can see that communication is accomplished with the normal I2C lines, which operate at 3.3V, but servo power is provided via schematic component J2, which is the barrel jack.
+
+This guide isn't meant to be a comprehensive power reference, but we just want to point out that if you have any components on your custom module that work outside of the 3.3V realm, you will [need to design for it][level_shifting]. To simplify your module design, we recommend using 3.3V components where possible.
+
+### The Power Warnings
+Here are some items to remember when working with power in electronics.
+
+  * **ALWAYS** unplug your Tessel and any external power before making or altering connections.
+  * Don't mix voltages unless you know what you're doing. For example, if you put 5V on any of the module pins, you can ruin your Tessel.
+  * Never connect the positive voltage directly to ground. This is called a [short circuit](http://en.wikipedia.org/wiki/Short_circuit) and can ruin components and your day.
+  * Always exercise caution and verify that you have hooked everything up correctly before plugging in your Tessel.
+
+## Communication
+
+Once you have decided how you are going to power your custom module, it's time to decide how the main Tessel board will talk to it.
+
+In the world of web communication, there are standards like HTTP, HTTPS, and FTP that allow different systems to talk to each other in well-defined ways. The same concept exists in the hardware world and the Tessel supports four standard communication protocols for talking to modules.
+
+  * [GPIO][comm_gpio]
+  * [SPI][comm_spi]
+  * [I2C][comm_i2c]
+  * [UART][comm_uart]
+
+Because the Tessel [does most of the heavy lifting][hardware_api] for all of these, you don't need to be an expert to use them in your custom module. However, if you'd like to learn a little more, we've [provided a simple overview of each](https://tessel.io/docs/communicationProtocols).
+
+### So Which Communication Protocol Should I Use?
+Knowing that there are four communication options available to you, which should you use for your custom module? The good news is that this will usually be decided for you based on the type of module you are creating. For example, most [PIR sensor modules][pir_project] will set a pin high when motion is detected, which can be read with a simple digital input (GPIO). The same applies to sensors. For example, the Si7005 temperature and humidity sensor on the [Climate Module][climate_module] communicates via the [I2C protocol][comm_i2c]. Usually sensors will only support one protocol– so the decision is easy, you use that one.
```

Comment:
Might as well update the reference of si7005 to si7020 since only produced a couple hundred si7005s.